### DISCUSSION TITLE
Aeotec Z-Stick 7 USB stick (ZWA010) and Aeotec Z-Pi 7 Raspberry Pi HAT/Shield (ZWA025) Z-Wave 700 Controllers with Z-Wave JS

### DISCUSSION THREAD

**Discussion by Hedda** [zwave-js/zwave-js#2549] [General] [answered]
FYI, Aeotec looks to have released both a USB stick/dongle and Raspberry Pi HAT/Shield compatible with Z-Wave Series 700 (Z-Wave Gen7).

Update! Compared to [Silabs UZB-7 USB Stick reference hardware](https://github.com/zwave-js/node-zwave-js/discussions/2544), these products by Aeotec does have RF region tuned/optimized antennas!

Aeotec Z-Stick 7 (ZWA010) is a Z-Wave USB stick/dongle Bridge Controller based on Silicon Labs ZGM130 module with an EFR32ZG14 SoC:

![image](https://user-images.githubusercontent.com/6320001/117120174-4c7d2600-ad93-11eb-9d5d-33269fcb40a1.png)

https://aeotec.com/z-wave-usb-stick/z-stick-7-vs-z-stick-gen5.html#buy

https://help.aeotec.com/support/solutions/folders/6000237130

https://help.aeotec.com/support/solutions/articles/6000230554-z-stick-7-technical-specifications-

https://aeotec.freshdesk.com/support/solutions/folders/6000237130

https://help.aeotec.com/support/solutions/articles/6000230552-z-stick-7-user-guide-

https://www.silabs.com/wireless/z-wave/700-series-modules

https://products.z-wavealliance.org/products/3854

Aeotec Z-Pi 7 (ZWA025) is a Z-Wave Raspberry Pi Shield Bridge Controller based on Silicon Labs ZGM130 module with an EFR32ZG14 SoC:

![image](https://user-images.githubusercontent.com/6320001/117119956-04f69a00-ad93-11eb-9c2f-3c712da9eeeb.png)

https://aeotec.com/z-wave-home-automation/development-kit-pcb.html

https://help.aeotec.com/support/solutions/articles/6000230553-z-pi-7-technical-specifications-

https://aeotec.freshdesk.com/support/solutions/folders/6000237129

https://help.aeotec.com/support/solutions/articles/6000230551-z-pi-7-user-guide-

https://aeotec.com/z-wave-gateways/comparison.html

https://www.silabs.com/wireless/z-wave/700-series-modem-soc

https://products.z-wavealliance.org/products/3853

Both run Silabs Z-Wave Series 700 (Z-Wave Gen7) stack with Serial API which should also be fully backwards with Z-Wave 500 Z-Wave Plus.

https://aeotec.com/z-wave-usb-stick/z-stick-7-vs-z-stick-gen5.html

https://aeotec.com/z-wave-gateways/comparison.html

Note! As I understand it. Z-Wave 700 MCU chips like the Silabs EFR32ZG14 SoC (and EFR32ZG14 SoC in ZGM130 module) are not hardcoded to a specific radio frequency or RF region.  EFR32ZG14 can modulate the RF frequency so the same chip part can support any sub-1GHz frequency can the RF region can be set via software as the RF region is configured when the gateway application initializes the Z-Wave network. Different Z-Wave 700 Controller adapter manufacturers may however choose to tune the physical antenna design for a specific radio frequency range in order to optimize it for a specific RF region. As such you could in theory maybe use the same part anywhere in the world as long as you reset it and set the correct RF region on initialization, but if it has an antenna design that has been tuned for a specific radio frequency range then it might not be optimal if used in a different region. For more information see https://github.com/zwave-js/node-zwave-js/issues/1674

PS: Notice they refer to Z-Pi7 as their "**_Aeotec-built development kit that lets you adapt existing hardware to work with Z-Wave_**".

**Comment by Hedda** [comment]
As I understand it, both of these should already work with Z-Wave JS in Z-Wave 500 / Z-Wave Gen5 backwards compatibility mode?

https://github.com/zwave-js/node-zwave-js/discussions/2134

https://aeotec.com/z-wave-gateways/compatibility.html#ZStick7

https://aeotec.com/z-wave-gateways/compatibility.html#ZPi7

AlCalzone wrote: 

"**_If you don't need any special functionality, you can use the 700 series sticks right now. In fact I am using one for development._**"

**Comment by AlCalzone** [comment]
What's the question here?

**Reply by Hedda** [reply]
> What's the question here?

Not a question. Posted under general discussion as an FYI.  As in, FYI, these new Z-Wave JS compatible adapters are now available. 

So just tp start an open general discussion and user experience feedback about using this hardware with Z-Wave JS.

PS: In case you wondering I'm just a home automation enthusiast who has no affiliation with Aeotec and Silabs or any resellers, etc..

**Reply by Hedda** [reply]
If you need a question to post and mark an answer to then, please let us know if you recommend any or both these as relatively future-proof hardware adapters for Z-Wave JS developers and/or Z-Wave JS advanced users and/or users who are new Z-Wave JS?

**Reply by Hedda** [reply]
Another question is about how Z-Wave Series 700 vs. Z-Wave Series 700 compatibility should be documented for Z-Wave JS?

https://aeotec.com/z-wave-gateways/compatibility.html#ZPi7

https://aeotec.com/z-wave-gateways/compatibility.html#ZStick7

Z-Wave JS integration for Home Assistant still do not list these under "Supported Z-Wave USB Sticks & Hardware Modules":

https://www.home-assistant.io/integrations/zwave_js#requirements HA's Z-Wave JS REQUIREMENTS still refers to: 

https://www.home-assistant.io/docs/z-wave/controllers/#supported-z-wave-usb-sticks--hardware-modules

and that is the old docs which still says: "**_USB sticks using the new 700 series Z-Wave platform are not compatible._** ".

Maybe 700 adapters should be listed but with comments about "experimental" and "limited to Z-Wave Series 500 support"?

**Reply by AlCalzone** [reply]
> Not a question

Alright.

> Z-Wave JS integration for Home Assistant still do not list these under "Supported Z-Wave USB Sticks & Hardware Modules":
> 
> [home-assistant.io/integrations/zwave_js#requirements](https://www.home-assistant.io/integrations/zwave_js#requirements) HA's Z-Wave JS REQUIREMENTS still refers to:
> 
> [home-assistant.io/docs/z-wave/controllers/#supported-z-wave-usb-sticks--hardware-modules](https://www.home-assistant.io/docs/z-wave/controllers/#supported-z-wave-usb-sticks--hardware-modules)

Please raise an issue in the HA repo for that. Z-Wave 700 controllers are compatible with Z-Wave JS, except for some advanced functionality like NVM backup/restore which isn't supported on the 500 series either.

**Reply by Hedda** [reply]
> > Not a question
> 
> Alright.
> 
> > Z-Wave JS integration for Home Assistant still do not list these under "Supported Z-Wave USB Sticks & Hardware Modules":
> > [home-assistant.io/integrations/zwave_js#requirements](https://www.home-assistant.io/integrations/zwave_js#requirements) HA's Z-Wave JS REQUIREMENTS still refers to:
> > [home-assistant.io/docs/z-wave/controllers/#supported-z-wave-usb-sticks--hardware-modules](https://www.home-assistant.io/docs/z-wave/controllers/#supported-z-wave-usb-sticks--hardware-modules)
> 
> Please raise an issue in the HA repo for that. Z-Wave 700 controllers are compatible with Z-Wave JS, except for some advanced functionality like NVM backup/restore which isn't supported on the 500 series either.

Should the Z-Wave JS project have a page in docs that list supported Z-Wave sticks/dongles that they could link to instead?

https://zwave-js.github.io/node-zwave-js/

https://github.com/zwave-js/node-zwave-js/tree/master/docs

https://github.com/zwave-js/node-zwave-js/tree/master/docs/getting-started

Maybe a dedicated page that lists supported Z-Wave sticks/dongles and specifications of those such as chip, specs, and FW?

Name | Remark | Interface | Antenna Type | Z-Wave Version | Long Range | Chip | Flash | RAM | Firmware
--  | -- | -- | -- | -- | -- | -- | -- | -- | --  
Aeotec Z-Stick Gen5 | -| - | - | - | - | - | - | - | - | -
Aeotec Z-Stick Gen5+ | -| - | - | - | - | - | - | - | - | -
Aeotec Z-Stick 7 | Model ZWA010 | - | - | - | - | - | - | - | - | -
Aeotec Z-Pi 7 | Model ZWA025 | - | - | - | - | - | - | - | - | -
Nortek/GoControl HUSBZB-1 | -| - | - | - | - | - | - | - | - | -
Sigma Designs UZB-5 | -| - | - | - | - | - | - | - | - | -
Sigma Designs UZB-7 | -| - | - | - | - | - | - | - | - | -
Zooz ZST10 500 Series | -| - | - | - | - | - | - | - | - | -
Zooz ZST10 700 Series | -| - | - | - | - | - | - | - | - | -
ZWave.me UZB1 | -| - | - | - | - | - | - | - | - | -
ZWave.me Razberry Board | -| - | - | - | - | - | - | - | - | -

**Reply by AlCalzone** [reply]
> specifications of those such as chip, specs, and FW?

That's gonna be impossible to keep updated. We could mention somewhere that it is compatible with all known 500 and 700 sticks though.

**Comment by Gamester17** [comment]
Aeotec Z-Stick 7 USB dongle (ZWA010) has been released for sale

https://popp.to/products/z-stick-7-zwa010

https://www.amazon.de/dp/B094VTQQCR/ref=asc_df_B094VTQQCR1622415600000/

https://www.amazon.co.uk/dp/B094VTQQCR/ref=asc_df_B094VTQQCR1622415600000/

Browse down to Z-Stick 7 on their page for USB sticks should get you to get links to other resellers

https://aeotec.com/z-wave-usb-stick/#buy

It looks like it will be available under different model numbers in different countries

* ZWA010-A = US Frequency 908,42 Mhz
* ZWA010-B: = AU Frequency 921,42 Mhz
* ZWA010-C = EU frequency 868,42 Mhz

**Comment by Hedda** [comment, marked as answer]
Aeotec now posted both a written guide and how-to to setup Aeotec Z-Stick 7 USB stick with [Z-Wave JS integration for Home Assistant](https://www.home-assistant.io/integrations/zwave_js/) here:

https://help.aeotec.com/support/solutions/articles/6000246295

https://www.youtube.com/watch?v=NKYGpwE4DOU

FYI, due to issue with network healing in shipped firmware it is recommended to upgrade Aeotec Z-Wave 700 Controller adapters firmware.

(See discussions regarding network heal issues due to bug in Z-Wave 700 Controller firmware versions prior to 7.15 here -> https://github.com/zwave-js/node-zwave-js/discussions/3319).

Tip! Best is probably to get recommended firmware version from Aeotec support as your first option, however, FYI, you can perform local firmware upgrade using OTW (Over The Wire) on any Z-Wave 700 USB sticks using the latest Z-Wave 700 Serial API firmware GBL XModem file via the "OTW Firmware Upgrade" option in Silabs "Z-Wave PC Controller" tool that gets installed with Simplicity Studio from Silabs.

https://aeotec.freshdesk.com/support/solutions/articles/6000250881-z-wave-command-class-configuration-tool-download-

https://aeotec.freshdesk.com/support/solutions/articles/6000247479-backup-and-restore-z-stick-7

### Quick Start-Up Guide to flash a Z-Wave 700 Controller with the latest firmware from Silabs

Here is a quick start-up guide on how to perform OTW Firmware Upgrade on ex. UZB-7 using "Z-Wave PC Controller" tool in Simplicity Studio.

Prerequisites: 

To actually get/download the latest firmware file for Z-Wave 700 USB stick you need to register for a free Silicon Labs Community account at https://community.silabs.com/SL_CommunitiesSelfReg and install Simplicity Studio v5 for your operating system, start it and login to install + update the Z-Wave technology type SDK package inside Simplicity Studio. See https://www.silabs.com/developers/simplicity-studio

1. Connect/plug-in the Z-Wave 700 Controller USB stick to your computer.
2. Z-Wave 700 Controller USB sticks will normally use a Silabs CP2102N USB-to-serial converter chip for the MCU SoC so if needed by your operating system then install the Virtual COM Port (VCP) drivers and CP210x VCP Drivers from Silicon Labs. The latest drivers are available at https://www.silabs.com/developers/usb-to-uart-bridge-vcp-drivers (for Windows 10 download and install the "CP210x Universal Windows Driver" package).
3. If using Windows 10 then be sure to disable Energy Saving Mode on the virtual com port so the dongle does not go into sleep: https://silabs-prod.adobecqms.net/community/wireless/z-wave/knowledge-base.entry.html/2020/06/04/uzb7_not_accessibleonwindows10-0Dw1
4. Start and run the "Z-Wave PC Controller" from Simplicity Studio from under the Tools menu.
5. Click on the “Setting-wheel” in the Right-Top corner of the Z-Wave PC Controller tool and select “Silicon Labs CP210x USB
to UART Bridge”. Click OK.
6. Click on "NVM Backup/Restore" select a folder and file path to backup files to in order to keep your a backup network stored. Make sure to name it with something like "ZWave_USB_Stick_Backup_date_and_time". Click on Backup.
7. Click the "OTW Firmware Upgrade" box and select the correct GBL file, (taking note that if you do not select a firmware with a version that is newer than your existing version you will get an error an upgrade will fail as downgrade or reflashing same version is not possible here). Firmware upgrade should be started and completed automatically.

FYI, if use default directories for Simplicity Studio + the Z-Wave technology type package GBL firmware binary files should after install be in:

C:\SiliconLabs\SimplicityStudio\v5\developer\sdks\gecko_sdk_suite\v3.2\protocol\z-wave\Apps\bin\gbl

Silabs UZB-7 USB Stick uses an EFR32ZG14 SoC so you need to select the latest ZW SerialAPI Controller firmware for it and your RF region.

So for example, currently, the latest firmware file (Version 7.16.2.0 / Version 7.16 Build 267) for the UZB7 USB sticks in Europe (RF region) is:

"ZW_SerialAPI_Controller_7.16.2_267_EFR32ZG14_REGION_EU.gbl"

PS: Move information is available in INS13114 Z-Wave PC based Controller v5 User Guide from Silicon Labs:

https://www.silabs.com/documents/public/user-guides/INS13114-Z-Wave-PC-Based-Controller-v5-User-Guide.pdf

**Comment by Hedda** [comment]
Aeotec now posted a 'how-to' upgrade firmware of their own and they also provide direct download links for the GBL firmware files for OTW:

https://help.aeotec.com/support/solutions/articles/6000252296-update-z-stick-7-sdk-7-15

**Comment by Hedda** [comment]
> Or is it best to just upgrade GPL firmware files via the mentioned Silicon Labs "Z-Wave PC Controller" application in Simplicity Studio.?
> 
> https://www.silabs.com/wireless/z-wave#software
> 
> https://www.silabs.com/community/wireless/z-wave/knowledge-base.entry.html/2018/12/19/z-wave_500_programm-BQRO
> 
> Or should one just upgrade firmware via bootloader boot mode XMODEM binary transfers using any Terminal program (like ex. Tera Term)?
> 
> https://community.silabs.com/s/article/z-wave-700-programming-uzb7-controller-stick?language=en_US
> 
> https://silabs-prod.adobecqms.net/content/usergenerated/asi/cloud/content/siliconlabs/en/community/wireless/z-wave/knowledge-base/jcr:content/content/primary/blog/z-wave_700_programm-4Hx0.social.0.10.html


FYI; blhoward2 confirmed OTW firmware upgrade bootloader + xmodem procedure in Linux on an UZB-7 stick as he posted this in https://github.com/zwave-js/node-zwave-js/issues/2545

> Confirmed manual process to upgrade working in Linux. Credit to cpw:
> 
> 1. installed minicom
> 2. figured out where device was (/dev/ttyUSB0 for me)
> 3. unplugged/plugged in device
> 4. in a shell: stty 115200 cs8 -cstopb -F /dev/ttyUSB0
> 5. then in same shell: printf '\x01\x03\x00\x27\xDB' >/dev/ttyUSB0
> 6. configure minicom: minicom -s important: turn hw flow control OFF here, set baud etc
> 7. run minicom: minicom -D /dev/ttyUSB0
> 8. press enter a bunch of  times, you'll see menu as demoed. (may need to replug stick to get this to happen)
> 9. select 1 in menu to flash. in minicom, use ctrl shift S, select xmodem, and specify path to file. Upload should proceed (it'll take a minute)
> 10. press 2 once upload is complete in minicom to "run" the new firmware.

**Comment by Hedda** [comment]
Great news to hear Aeotec Z-Pi 7 will be compatible with GPIO pins and enclosure for “Home Assistant Amber” / “Home Assistant Yellow”:

https://www.crowdsupply.com/nabu-casa/home-assistant-yellow/updates/z-wave-and-rename

![image](https://user-images.githubusercontent.com/6320001/143401380-ab664d22-69de-4ad9-8296-2ef391899614.png)

### _Z-Wave and Rename_

_Today we are announcing the rename to Home Assistant Yellow. It will stay the same product we announced as Home Assistant Amber, it’s just a new name._

_A couple of weeks back Lewis from Everything Smart Home asked me if Aeotec’s Z-Pi 7 would work with Home Assistant Yellow. We have a subset of the GPIO pins available on the J11 header, which offers plenty of 3.3 V power and a UART directly connected to the Compute Module 4. Electronically, this is all what is required to make the Z-Pi 7 work._

_We received two Z-Pi 7 from our friends at Aeotec and can confirm: Z-Pi 7 works with Home Assistant Yellow and the Z-Wave JS Add-on and Integration. It also fits into the case which provides an easy and clean way to extend Home Assistant Yellow with Z-Wave support._

_We haven’t measured or compared RF performance yet. The orientation of the antenna might not be quite ideal, but keep in mind that Z-Wave forms a mesh-network. Z-Wave also uses a different frequency spectrum than the onboard Zigbee radio, hence interference should be minimal._

![image](https://user-images.githubusercontent.com/6320001/143401397-f1859fd7-4e23-4c5e-b937-617f141e0c45.png)

**Reply by AseKarlsson** [reply]
Anyone found/made a simple guide on how to upgrade the Firmware of Z-Pi 7 on HA Yellow?
Those I found is not that straight forward ☹

**Comment by Hedda** [comment]
FYI, here is direct links for EU and US regions to 7.17.2 firmware image download for Aeotec Z-Stick 7 and Aeotec Z-Pi 7 (based on EFR32ZG14):

https://github.com/SiliconLabs/gecko_sdk/blob/gsdk_4.0/protocol/z-wave/Apps/bin/gbl/ZW_SerialAPI_Controller_7_17_2_406_EFR32ZG14_REGION_EU.gbl?raw=true

https://github.com/SiliconLabs/gecko_sdk/blob/gsdk_4.0/protocol/z-wave/Apps/bin/gbl/ZW_SerialAPI_Controller_7_17_2_406_EFR32ZG14_REGION_US.gbl?raw=true

As mentioned before, Silicon Labs are now releasing on GitHub so other regions and different chips can also be found by selecting "Go to file"  on https://github.com/SiliconLabs/gecko_sdk/tree/gsdk_4.0/protocol/z-wave/Apps/bin/gbl and then entering "`ZW_SerialAPI_Controller_7_17_2_406_`" (for web download must then go to the GiHub web page first and then click "Download").

**Reply by jaspweowaale** [reply]
I received the US version of the Z-Stick Gen7 ZWA010 by mistake and I am currently in Vietnam. It took over five weeks to get here, and I'm trying to avoid another long wait. Does anyone know if it's possible to reconfigure the US version to operate on EU frequencies through software or firmware updates? Any advice would be greatly appreciated!

Thank you!

Reding this,  but need a tip to move from webpage to my PI or Linix machine setup 

https://github.com/zwave-js/node-zwave-js/issues/1674

**Reply by Hedda** [reply]
Aeotec makes three different models for a reason, and for Asia Pacific RF regions you should have bought their ”ZWA010-B (AU)” model which board have hardware filters for the antenna that has specifically been tuned for frequency 921,42 Mhz instead of the ”ZWA010-A (US)” model that has antenna hardware filters tuned for frequency 908,42 Mhz. So all their board uses hardware filters that different tuning to filter out frequencies outside the regions that they were made for.

https://www.silabs.com/wireless/z-wave/global-regions

You should contact Aeotec for a definitive answer, but even though you can change the RF region in software I am willing to bet that Aeotec will say that changing from USA to AU freqencies is not recommend and that you should expect to get a bad experince if you change the RF region on a adapter that has specifically tunned hardware filters such as these have.

https://aeotec.freshdesk.com/support/solutions/articles/6000230554-z-stick-7-technical-specifications-

https://aeotec.freshdesk.com/support/solutions/articles/6000230552-z-stick-7-user-guide-

My guess it that it will probably still work if you change the RF region in software but due to the hardware filters for the antenna the reception will be poor which affect the range, so range will be signifigantly impaired that it will probably not be worth it, as you probably need to add many repeater devices very close to the controller to get any reception.

**Comment by jaspweowaale** [comment]
Looking at the Chips, they all have the same model just the radio is set
differently, I spent a few hours on this and found a lot of wrong info and
lots of dead links also in the end, what worked was the below,

The Silicon Labs development tool lets you set Frq on the radio, and yes
its works, I needed EU as all my z-wave is from Eur, and I don't see any different in output btw or range 

You will need a Silicon Labs account and download a few things, but as
02-06-2024 this works

https://aeotec.freshdesk.com/support/solutions/articles/6000252296-update-z-stick-7-sdk-7-15

https://aeotec.freshdesk.com/support/solutions/articles/6000226205-z-wave-command-class-configuration-tool-download-

![image](https://github.com/zwave-js/node-zwave-js/assets/66903625/38a9cb82-108d-410b-b9d9-d1cc187e5de7)





Best regards.

Jasper O. Waale,
